### PR TITLE
feat(flag): Remove internal check from trends

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
Undoing this now that it is no longer necessary

### CODE REVIEW

File: src/sentry/static/sentry/app/views/performance/landing.tsx [hunk 1]
```diff
@@ -260,7 +260,7 @@ class PerformanceLanding extends React.Component<Props, State> {
 
   renderHeaderButtons() {
     return (
-      <Feature features={['trends', 'internal-catchall']} requireAll={false}>
+      <Feature features={['trends']} requireAll={false}>
```
Comment (wmak) [inline src/sentry/static/sentry/app/views/performance/landing.tsx:263]:
```suggestion
      <Feature features={['trends']}>
```

Comment (dashed) [inline src/sentry/static/sentry/app/views/performance/landing.tsx:263]: `requireAll` needs to be omitted

File: src/sentry/static/sentry/app/views/performance/trends/content.tsx [hunk 2]
```diff
@@ -97,7 +97,7 @@ class TrendsContent extends React.Component<Props, State> {
     const showChangedProjects = hasMultipleProjects(selection);
 
     return (
-      <Feature features={['trends', 'internal-catchall']} requireAll={false}>
+      <Feature features={['trends']} requireAll={false}>
```
Comment (wmak) [inline src/sentry/static/sentry/app/views/performance/trends/content.tsx:100]:
```suggestion
      <Feature features={['trends']}>
```

**Review by wmak — APPROVED** [review]
1 nit, but LGTM

File: src/sentry/static/sentry/app/views/performance/landing.tsx (hunk 1 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/views/performance/landing.tsx:263]: oops, you're right.

**Review by dashed — APPROVED** [review]

**Pull request merged**